ollama/_client.py

```diff
@@ -450,126 +524,184 @@ def _create_blob(self, path: Union[str, Path]) -> str:
     digest = f'sha256:{sha256sum.hexdigest()}'
 
     try:
-      self._request('HEAD', f'/api/blobs/{digest}')
+      self._request_raw('HEAD', f'/api/blobs/{digest}')
     except ResponseError as e:
       if e.status_code != 404:
         raise
 
       with open(path, 'rb') as r:
-        self._request('POST', f'/api/blobs/{digest}', content=r)
+        self._request_raw('POST', f'/api/blobs/{digest}', content=r)
 
     return digest
 
-  def delete(self, model: str) -> Mapping[str, Any]:
-    response = self._request('DELETE', '/api/delete', json={'name': model})
-    return {'status': 'success' if response.status_code == 200 else 'error'}
+  def list(self) -> ListResponse:
+    return self._request(
+      ListResponse,
+      'GET',
+      '/api/tags',
+    )
 
-  def list(self) -> Mapping[str, Any]:
-    return self._request('GET', '/api/tags').json()
+  def delete(self, model: str) -> StatusResponse:
+    r = self._request_raw(
+      'DELETE',
+      '/api/delete',
+      json=DeleteRequest(
+        model=model,
+      ).model_dump(exclude_none=True),
+    )
+    return StatusResponse(
+      status='success' if r.status_code == 200 else 'error',
+    )
 
-  def copy(self, source: str, destination: str) -> Mapping[str, Any]:
-    response = self._request('POST', '/api/copy', json={'source': source, 'destination': destination})
-    return {'status': 'success' if response.status_code == 200 else 'error'}
+  def copy(self, source: str, destination: str) -> StatusResponse:
+    r = self._request_raw(
+      'POST',
+      '/api/copy',
+      json=CopyRequest(
+        source=source,
+        destination=destination,
+      ).model_dump(exclude_none=True),
+    )
+    return StatusResponse(
+      status='success' if r.status_code == 200 else 'error',
+    )
 
-  def show(self, model: str) -> Mapping[str, Any]:
-    return self._request('POST', '/api/show', json={'name': model}).json()
+  def show(self, model: str) -> ShowResponse:
+    return self._request(
+      ShowResponse,
+      'POST',
+      '/api/show',
+      json=ShowRequest(
+        model=model,
+      ).model_dump(exclude_none=True),
+    )
 
-  def ps(self) -> Mapping[str, Any]:
-    return self._request('GET', '/api/ps').json()
+  def ps(self) -> ProcessResponse:
+    return self._request(
+      ProcessResponse,
+      'GET',
+      '/api/ps',
+    )
```

Comment:
Need the model dump here as well
```suggestion
    ).model_dump(exclude_none=True)
```

Reply:
Going to tackle these comments in new PR